README.md

```diff
@@ -1304,113 +1406,1033 @@ public:
+
+- `RawPointerList`
+- `RawPointerQueue`
+- `RawPointerTransactionalMultiMap`
+- `RawPointerMap`
+- `RawPointerSet`
+- `RawPointerTransactionalQueue`
+- `RawPointerMultiMap`
+- `RawPointerTransactionalMap`
+
+These are adapter classes and they do not create new structures. You just provide the legacy containers as parameters and then you can work with these raw capability containers freely. An example usage of `RawPointerMap` is shown below:
+
+```
+hazelcast::client::IMap<std::string, std::string> m = hz.getMap<std::string, std::string>("map");
+hazelcast::client::adaptor::RawPointerMap<std::string, std::string> map(m);
+map.put("1", "Tokyo");
+map.put("2", "Paris");
+map.put("3", "New York");
+std::cout << "Finished loading map" << std::endl;
+
+std::auto_ptr<hazelcast::client::DataArray<std::string> > vals = map.values();
+std::auto_ptr<hazelcast::client::EntryArray<std::string, std::string> > entries = map.entrySet();
+
+std::cout << "There are " << vals->size() << " values in the map" << std::endl;
+std::cout << "There are " << entries->size() << " entries in the map" << std::endl;
+
+for (size_t i = 0; i < entries->size(); ++i) {
+   const std::string * key = entries->getKey(i);
+      if ((std::string *) NULL == key) {
+			std::cout << "The key at index " << i << " is NULL" << std::endl;
+        } else {
+            std::auto_ptr<std::string> val = entries->releaseValue(i);
+            std::cout << "(Key, Value) for index " << i << " is: (" << *key << ", " <<
+                (val.get() == NULL ? "NULL" : *val) << ")" << std::endl;
+        }
+    }
+    std::cout << "Finished" << std::endl;
+```
+
+Raw pointer API uses the DataArray and EntryArray interfaces which allow late deserialization of objects. The entry in the returned array is deserialized only when it is accessed. Please see the example code below:
+
+```
+// No deserialization here
+std::auto_ptr<hazelcast::client::DataArray<std::string> > vals = map.values(); 
+
+// deserializes the item at index 0 assuming that there are at least 1 items in the array
+const std::string *value = vals->get(0);
+
+// no deserialization here since it was already de-serialized
+value = vals->get(0);
+
+// no deserialization here since it was already de-serialized
+value = (*vals)[0];
+
+// releases the value so that you can keep this object pointer in your application at some other place
+std::auto_ptr<std::string> releasedValue = vals->release(0);
+
+// deserialization occurs again since the value was released already
+value = vals->get(0);
+```
+
+Using raw pointer based API may improve performance if you are using the API to return multiple values such as values, keySet, and entrySet. In this case, cost of deserialization is delayed until the item is actually accessed.
```

Comment:
Using the raw pointer based API may improve the performance if you are using the API to return multiple values such as `values`, `keySet` and `entrySet`. In this case, the cost of deserialization is delayed until the item is actually accessed.